ocean_provider/requests.py

```diff
@@ -0,0 +1,181 @@
+from eth_utils import add_0x_prefix
+from flask import request as flask_request
+from flask_sieve import JsonRequest
+from flask_sieve.rules_processor import RulesProcessor
+from flask_sieve.validator import Validator
+from ocean_utils.did import did_to_id
+
+from ocean_provider.access_token import AccessToken
+from ocean_provider.exceptions import InvalidSignatureError
+from ocean_provider.user_nonce import UserNonce
+from ocean_provider.util import get_request_data
+from ocean_provider.utils.accounts import verify_signature
+from ocean_provider.utils.basics import get_config
+
+user_nonce = UserNonce(get_config().storage_path)
+user_access_token = AccessToken(get_config().storage_path)
+
+
+class CustomJsonRequest(JsonRequest):
+    def __init__(self, request=None):
+        request = request or flask_request
+        request = get_request_data(request)
+        self._validator = CustomValidator(rules=self.rules(), messages={
+            'signature.signature': 'Invalid signature provided.',
+            'transferTxId.access_token': 'There is already a token with these parameters'  # noqa
+        }, request=request)
+
+
+class CustomValidator(Validator):
+    def __init__(
+        self, rules=None, request=None, custom_handlers=None, messages=None,
+        **kwargs
+    ):
+        super(CustomValidator, self).__init__(
+            rules, request, custom_handlers, messages, **kwargs
+        )
+        self._processor = CustomRulesProcessor()
+
+
+class CustomRulesProcessor(RulesProcessor):
+    def validate_signature(self, value, params, **kwargs):
+        self._assert_params_size(size=3, params=params, rule='signature')
+        owner = self._attribute_value(params[0])
+        did = self._attribute_value(params[1])
+        rule = params[2]
+        original_msg = f'{owner}{did}' if rule == 'customer_did' else f'{did}'
```

Comment:
what is `customer_did`?

Reply:
the rule for the original message to check in the signature: it is either customer_address + did OR just the did. This is a string defining the customer_address + did rule.

Reply:
i think we use `consumer` typically

Reply:
done ✅